dynacred/src/group/groupContractCollection.ts

```diff
@@ -0,0 +1,341 @@
+import { Private } from "../KeyPair";
+import { GroupContract } from "./groupContract";
+import { GroupDefinition } from "./groupDefinition";
+
+/**
+ * Class collecting group contracts from the same group
+ */
+export class GroupContractCollection {
+
+    static fromObject(obj: any) {
+        const gcCollection = new GroupContractCollection();
+
+        if (obj.collection) {
+            Object.keys(obj.collection).forEach((id) => {
+                gcCollection._collection.set(id,
+                    obj.collection[id].map((gc: any) => GroupContract.fromObject(gc)),
+                );
+            });
+        }
+
+        if (obj.currentGroupContract) {
+            gcCollection.currentGroupContract = GroupContract.fromObject(obj.currentGroupContract);
+        }
+
+        if (obj._purpose) {
+            gcCollection._purpose = obj._purpose;
+        }
+
+        return gcCollection;
+    }
+
+    private _collection: Map<string, GroupContract>; // key: contractID, value: GroupDefinition
+    private _purpose: string;
+    private currentGroupContract: GroupContract;
+
+    constructor(purpose?: string) {
+        this._collection = new Map();
+        this._purpose = purpose ? purpose : undefined;
+    }
+
+    /**
+     * Create new group contract
+     *
+     * @param parent new group contract's predecessor
+     * @param groupDefinition new group contract's group definition
+     * @returns {GroupContract} new group contract
+     */
+    createGroupContract(parent: GroupContract, groupDefinition: GroupDefinition): GroupContract {
+        try {
+            let newGroupContract: GroupContract;
+            if (parent) {
+                newGroupContract = parent.proposeGroupContract(groupDefinition);
+            } else {
+                newGroupContract = new GroupContract(groupDefinition);
+            }
+            this.append(newGroupContract);
```

Comment:
I do not understand why it is an unexpected behavior: creating a new group contract automatically append it to the collection of group contract.

Reply:
Creating smth is different that appending smth, the name of the function/comments should reflect that; in general, there should be one and only way to code a code interaction. Otherwise, someone (like me) would create a `GroupContact` using `createGroupContract` then `append` it to the collection, and from a semantic point of view, it would *read* correclty. Being explicit about what is appending avoid bugs in the long (and short) run.